drivers/DeepSmart/deepsmart/src/deepsmart/wisers.lua

```diff
@@ -0,0 +1,834 @@
+  end,
+  'wiser loop')
+  log.info('add wisers over')
+end
+------------
+-- del device
+------------
+function Wisers.del_device_by_id(devid)
+  local id = Wisers.idmap[devid]
+  if (id == nil) then
+    log.warn('dev '..devid..' is not in hub')
+    return false
+  end
+  log.info('dev '..devid..' hub id '..id..' delete from hub')
+  Wisers.driver:try_delete_device(id)
+  return true
+end
+function Wisers.del_device(device)
+  local parent_assigned_child_key = device.parent_assigned_child_key
+  log.info('del device device_network_id '..device.device_network_id)
+  if (parent_assigned_child_key == nil) then
+    log.info('del bridge '..device.device_network_id)
+    Wisers.del_wiser(device.device_network_id)
+    Wisers.driver:try_delete_device(device.id)
+  else
+    -- del device from hub
+    log.info('del device '..device.id)
+    Wisers.driver:try_delete_device(device.id)
+    local dev_id,_,idx = Wisers.parse_device_network_id(parent_assigned_child_key)
+    log.trace('del device '..parent_assigned_child_key..' -> dev_id '..dev_id..' idx '..idx)
+    -- get device wiser
+    local wiser,_ = Wisers.get_device_wiser(dev_id)
+    if (wiser == nil) then
+      log.warn('device '..dev_id..' is not exist for del device')
+      return false,'wiser is nil'
+    end
+    -- del config dev
+    if (wiser.config ~= nil and wiser.config.devices ~= nil) then
+      wiser.config.devices[dev_id] = nil
+    end
+  end
+end
+------------
+-- unbind the wiser
+-- it's not used now(no scene to use)
+------------
+function Wisers.del_wiser(wiser_index_code)
+  local wiser = Wisers.get_wiser(wiser_index_code)
+  if (wiser == nil) then
+    return false
+  end
+  wiser.state = 0
+  Wisers.wisers[wiser_index_code] = nil
+  log.info('wiser '..wiser_index_code..' del over')
+  return true
+end
+---------------
+-- get device wiser
+-- returns
+--    wiser,device
+---------------
+function Wisers.get_device_wiser(dev_id)
+  -- check all wisers to find the device
+  for i,wiser in pairs(Wisers.wisers) do
+    if (wiser.config ~= nil and wiser.config.devices ~= nil) then
+      local device = wiser.config.devices[dev_id]
+      if (device ~= nil) then
+        return wiser,device
+      end
+    end
+  end
+  return nil,nil
+end
+
+
+-- reload devices&&dps
+function Wisers.reload(uuid, add_devs, del_devs)
+  local wiser = Wisers.get_wiser(uuid)
+  if (wiser == nil or wiser.bridge == nil) then
+    log.warn('wiser '..uuid..' is not exist for reload')
+    return true
+  end
+  log.info('reload wiser '..wiser.wiser_index_code..' ip('..(wiser.ip or '')..')')
+  -- if wiser ip is not changed then do nothing
+  local deepsmartapi = wiser.api
+  if (deepsmartapi == nil) then
+    log.trace('wiser api is not inited')
+    return false
+  end
+  -- get devices
+  local configstr = deepsmartapi:load_config()
+  log.trace(string.format('load config (%s)', configstr))
+  if (configstr ~= nil and configstr ~= wiser.bridge:get_field(config.FIELD.DEVICES)) then
```

Comment:
if configstr is nil means load config fail, do not update devices(nil config will clear all devices)

Reply:
Should this `return false` if the configstr is nil? I dont think this function is "successful" if that fails, but I could be misunderstanding.

Reply:
Change to return false implementation